### PR TITLE
add support to run db tests from engine

### PR DESCRIPTION
**Pull request by PrateekGarg-gs**
depends on legend-pure changes - [pull request](https://github.com/finos/legend-pure/pull/538)

### CODE REVIEW

File: legend-engine-server/src/test/java/org/finos/legend/engine/server/test/pureClient/stores/dbSpecific/Test_Relational_DbSpecific_UsingPureClientTestSuite.java [hunk 1]
```diff
@@ -14,31 +14,57 @@
...
-                        return suite;
-                    },
-                    false, testServerConfigFilePath);
+    private static final ThreadLocal<ServersState> state = new ThreadLocal<>();
+
+    public static Test createSuite( String dbType, String dbTestCollectionPath, String testServerConfigFilePath ) throws Exception
+    {
+        CompiledExecutionSupport executionSupport = getClassLoaderExecutionSupport();
+
+        try
```
Comment (vikask-gs) [inline legend-engine-server/src/test/java/org/finos/legend/engine/server/test/pureClient/stores/dbSpecific/Test_Relational_DbSpecific_UsingPureClientTestSuite.java:37]: unnecessary try stmt

Comment (PrateekGarg-gs) [reply legend-engine-server/src/test/java/org/finos/legend/engine/server/test/pureClient/stores/dbSpecific/Test_Relational_DbSpecific_UsingPureClientTestSuite.java]: removed

**Comment by vikask-gs** [comment]
Looks good

**Comment by PrateekGarg-gs** [comment]
will raise a new PR to include corresponding old changes in legend-pure

**Comment by PrateekGarg-gs** [comment]
raised new PR to accommodate base code changes from legend-pure to legend-engine 
https://github.com/finos/legend-engine/pull/671

**Pull request closed without merging**